docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,346 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+## Introduction
+
+Breadth-first search (BFS) is an algorithm to traverse/search the data structure tree or a graph. It is similar to level-order traversal in trees. It starts at the root node and explores all of the neighbor nodes at the present node before moving on to the nodes at the next depth level. BFS uses a **'queue'** data structure for traversing the nodes.
+
+
+## Explanation
+
+- The BFS traversal of the graph is similar to the level-order traversal in trees. It starts the traversal with a given `source vertex (V)`, visits all of the vertices adjacent to the given vertex and `pushes them all to a queue` in order of visiting and `pops it from the front of the queue`, visits all adjacent nodes and pushes them into the queue. This process is repeated until the queue is empty or all of the vertices are visited.
+#### Quick Example :  
+- Consider the graph shown below. The vertex numbered 0 is the source vertex i.e. the BFS traversal will start from the vertex 0.
+
+**Step 1]**![BFS 1](https://user-images.githubusercontent.com/72685035/115226603-0ede9300-a12d-11eb-81f6-7947fa9a3366.jpg) 
+
+**Step 2]**![BFS 2](https://user-images.githubusercontent.com/72685035/115226742-36cdf680-a12d-11eb-83c9-018e1654a0fb.jpg) 
+
+**Step 3]**![BFS 3](https://user-images.githubusercontent.com/72685035/115226830-4cdbb700-a12d-11eb-8fe2-37987f6d1ff1.jpg) 
+
+**Step 4]**![BFS 4](https://user-images.githubusercontent.com/72685035/115226837-5107d480-a12d-11eb-89ca-8ce70a1c3583.jpg)
+
+**Step 5]**![BFS 5](https://user-images.githubusercontent.com/72685035/115226850-549b5b80-a12d-11eb-9c48-bccb2a9a831c.jpg)
```

Comment:
Did you make these diagrams?

Reply:
No, but they are not copied either, so they can be included